web/src/search_pill.ts

```diff
@@ -150,18 +150,21 @@ export function generate_pills_html(suggestion: Suggestion, query: string): stri
             // We can simply differentiate these cases by checking if `:` is
             // present in the query. See `set_search_bar_contents` for more
             // context.
-            if (query.includes(":")) {
+            //
+            // Note: We only want to do this check for the last search term
+            // which corresponds to the currently typed query.
```

Comment:
Can this note explain why? I also wonder if the other comments above should be revised.